unix-process/main.c

```diff
@@ -0,0 +1,119 @@
+#include <stdio.h>
+#include <unistd.h>
+#include <sys/wait.h>
+#include "employee.h"
+#include <stdlib.h>
+
+pid_t fork(void);
+
+void read_and_print_from_binary(int num_of_queries, FILE* input)
+{
+	for (int i = 0; i < num_of_queries; i++) {
+		struct employee emp;
+		fread(&emp, sizeof(struct employee), 1, input);
+
+		printf("%d %s %lf\n", emp.num, emp.name, emp.hours);
+	}
+}
+
+void procces_work(const char* procces_name, char* params[])
+{
+	pid_t pid = vfork();
+
+	switch (pid)
+	{
+	case -1:
+	{
+		fprintf(stderr, "Unable to fork\n");
+	}
+	case 0:
+	{
+		if (execv(procces_name, params) == -1)
+		{
+			fprintf(stderr, "Unable to exec\n");
+			break;
+		}
+	}
+	default:
+	{
+		int status;
+		waitpid(pid, &status, 0);
+		int exit_code = WEXITSTATUS(status);
```

Comment:
why don't you return it from this function? why do you continue execution after unsuccessful return code?